tests/test_coordinate_source.py

```diff
@@ -0,0 +1,277 @@
+        )
+        self.files_star = list(
+            zip(self.all_mrc_paths, sorted(glob(self.data_folder + "/sample*.star")))
+        )
+        self.files_box_nonsquare = list(
+            zip(self.all_mrc_paths, sorted(glob(self.data_folder + "/nonsquare*.box")))
+        )
+
+    def tearDown(self):
+        self.tmpdir.cleanup()
+
+    def testLoadFromCoord(self):
+        # ensure successful loading from particle center files (.coord)
+        CentersCoordinateSource(self.files_coord, particle_size=256)
+
+    def testLoadFromStar(self):
+        # ensure successful loading from particle center files (.star)
+        CentersCoordinateSource(self.files_star, particle_size=256)
+
+    def testLoadFromRelionAutoPick(self):
+        # Here we test loading from a Relion project directory's STAR
+        # index file. We are not using the simulated data source constructed
+        # in the setup
+        RelionCoordinateSource(
+            os.path.join(
+                self.test_dir_root, "AutoPick/job006/sample_relion_autopick.star"
+            ),
+            particle_size=256,
+        )
+
+    def testNonSquareParticles(self):
+        # nonsquare box sizes must fail
+        with self.assertRaises(ValueError):
+            EmanCoordinateSource(self.files_box_nonsquare)
+
+    def testOverrideParticleSize(self):
+        # it is possible to override the particle size in the box file
+        src_new_size = EmanCoordinateSource(self.files_box, particle_size=100)
+        src_from_centers = CentersCoordinateSource(self.files_coord, particle_size=100)
+        imgs_new_size = src_new_size.images(0, 10)
+        imgs_from_centers = src_from_centers.images(0, 10)
+        for i in range(10):
+            self.assertTrue(np.array_equal(imgs_new_size[i], imgs_from_centers[i]))
+
+    def testImages(self):
+        # load from both the box format and the coord format
+        # ensure the images obtained are the same
+        src_from_coord = CentersCoordinateSource(self.files_coord, particle_size=256)
+        src_from_star = CentersCoordinateSource(self.files_star, particle_size=256)
+        src_from_relion = RelionCoordinateSource(
+            relion_autopick_star=os.path.join(
+                self.test_dir_root, "AutoPick/job006/sample_relion_autopick.star"
+            ),
+            particle_size=256,
+        )
+        imgs_box = self.src_from_box.images(0, 10)
+        imgs_coord = src_from_coord.images(0, 10)
+        imgs_star = src_from_star.images(0, 10)
+        imgs_relion = src_from_relion.images(0, 10)
+        for i in range(10):
+            self.assertTrue(np.array_equal(imgs_box[i], imgs_coord[i]))
+            self.assertTrue(np.array_equal(imgs_coord[i], imgs_star[i]))
+            self.assertTrue(np.array_equal(imgs_star[i], imgs_relion[i]))
+
+    def testImagesRandomIndices(self):
+        # ensure that we can load a specific, possibly out of order, list of
```

Comment:
This is good, but I think probably either of the two randomized methods would suffice.  If you wanted to be sure you got coverage from both micrographs just take a random sample from :220 and 220: then shuffle those, or something.

Reply:
Removed the first one, I agree. Thanks